src/jit/valuenum.cpp

```diff
@@ -1630,6 +1631,27 @@ double ValueNumStore::GetConstantDouble(ValueNum argVN)
     return result;
 }
 
+// Given a float constant value number return its value as a float.
+//
+float ValueNumStore::GetConstantSingle(ValueNum argVN)
+{
+    assert(IsVNConstant(argVN));
+    var_types argVNtyp = TypeOfVN(argVN);
+    assert(argVNtyp == TYP_FLOAT);
+
+    float result = 0;
+
+    switch (argVNtyp)
```

Comment:
`argVNTyp` is only used in an assert and in a single-case switch with an `unreached` default case, which effectively asserts the same thing again.  I'd suggest getting rid of `argVNTyp` (use `TypeOfVN(argVN)` in the assertion) and simplifying away the switch.